gnat2goto/driver/tree_walk.adb

```diff
@@ -4299,99 +4325,120 @@ package body Tree_Walk is
         (LHS_Element_Type, RHS_Element_Type, Index_Type);
       Map_Cursor : Array_Copy_Maps.Cursor;
       Map_Inserted : Boolean;
-   begin
-      Array_Copy_Map.Insert (Map_Key, Ireps.Empty, Map_Cursor, Map_Inserted);
-      if not Map_Inserted then
-         return Array_Copy_Maps.Element (Map_Cursor);
-      end if;
 
-      --  Create a new copy function:
-      declare
-         Func_Type : constant Irep := New_Irep (I_Code_Type);
-         Func_Args : constant Irep := New_Irep (I_Parameter_List);
-         Write_Ptr_Arg : constant Irep := New_Irep (I_Code_Parameter);
-         Read_Ptr_Arg : constant Irep := New_Irep (I_Code_Parameter);
+      function Build_Copy_Function return Symbol;
+
+      -------------------------
+      -- Build_Copy_Function --
+      -------------------------
+
+      --  Build a functions:
+      --  n.b.: the typecast is optional
+      --  void copy_array(lhs_type* out, rhs_type* in, int len) {
```

Comment:
I've been wondering - isn't this basically just `memcpy`? We do model that, for instance a C-program like this:

```
#include <string.h>
#include <assert.h>
#include <stdlib.h>

int main(void) {
  const int array_size = 10;
  int *arr = malloc(sizeof(int) * array_size);
  int *arr_copy = malloc(sizeof(int) * array_size);
  for(int i = 0; i < array_size; ++i) {
    arr[i] = i;
  }
  memcpy(arr_copy, arr, sizeof(int)*array_size);
  for(int i = 0; i < array_size; ++i) {
    assert(arr_copy[i] == i);
  }
}
```

can be verified just fine. I understand you were having troubles with this function, so maybe that can be mitigated this way? (i.e. you should be able to just call it the same way you deal with `malloc`)

Reply:
will try, ta.